packages/@react-spectrum/dialog/stories/Dialog.stories.tsx

```diff
@@ -350,31 +350,50 @@ function renderWithForm({width = 'auto', ...props}) {
+    <p>Cras semper auctor neque vitae tempus quam pellentesque nec. Maecenas ultricies mi eget mauris pharetra et ultrices neque ornare. Vulputate enim nulla aliquet porttitor lacus luctus accumsan tortor posuere. Pellentesque habitant morbi tristique senectus et. Ipsum dolor sit amet consectetur adipiscing elit pellentesque. Sem et tortor consequat id porta nibh venenatis. Viverra nibh cras pulvinar mattis nunc sed blandit. Urna porttitor rhoncus dolor purus. Vivamus arcu felis bibendum ut. Cras sed felis eget velit aliquet. Sed tempus urna et pharetra pharetra. Viverra adipiscing at in tellus integer feugiat scelerisque varius morbi. Ac tortor vitae purus faucibus ornare suspendisse sed nisi lacus. Ultrices neque ornare aenean euismod elementum nisi quis eleifend quam. Vel turpis nunc eget lorem. Quisque egestas diam in arcu cursus euismod quis viverra. At tempor commodo ullamcorper a lacus vestibulum sed. Id aliquet lectus proin nibh nisl condimentum id venenatis. Quis viverra nibh cras pulvinar. Purus in mollis nunc sed.</p>
   </React.Fragment>
 );
 
 function renderLongContent({width = 'auto', ...props}) {
   return (
-    <div style={{display: 'flex', width, margin: '100px 0'}}>
-      <DialogTrigger defaultOpen>
+    <div style={{display: 'flex', flexDirection: 'column', alignItems: 'start', width, padding: '0 20px'}}>
+      <p>
+        Test instructions on mobile:
+        <ol>
+          <li>Scroll down and open dialog.</li>
+          <li>Tap on the top input. Dialog should resize when the keyboard comes up and remain at the top of the screen. The rest of the page should not scroll (the first time you do this it may scroll slightly due to the bottom toolbar showing).</li>
+          <li>Close the software keyboard.</li>
+          <li>Scroll down and tap on the bottom input. Dialog should resize as before, and focused input should scroll into view.</li>
+          <li>Tap the previous button in the software keyboard. Focus should move to the first input in the dialog. Dialog should not move and page should not scroll.</li>
+          <li>Tap the previous button again. Focus should stay on the first input in the dialog and not move out to the textfield outside the dialog. The page may jump slightly, and come back but there is nothing we can do about this.</li>
+        </ol>
+        Note: all this only works outside the storybook iframe due to VisualViewport limitations.
```

Comment:
is this going to be problematic in ExC shell?

Reply:
yes, but there is nothing we can do about it 😢. we'd need to discuss with them how to mount overlays outside the iframe. luckily I don't think shell has many mobile users...

Reply:
😢